tests/unit/hive_metastore/test_locations.py

```diff
@@ -99,3 +109,61 @@ def test_external_locations():
     assert result_set[4].location == "jdbc:mysql://somemysql.us-east-1.rds.amazonaws.com:3306/test_db"
     assert result_set[5].location == "jdbc:providerknown://somedb.us-east-1.rds.amazonaws.com:1234/test_db"
     assert result_set[6].location == "jdbc:providerunknown://somedb.us-east-1.rds.amazonaws.com:1234/test_db"
+
+
+def make_row(data, columns):
+    row = Row(data)
+    row.__columns__ = columns
+    return row
+
+
+def test_save_external_location_mapping_missing_location():
+    ws = MagicMock()
+    select_cols = ["location", "storage_properties"]
+    sbe = MockBackend(
+        rows={
+            "SELECT location, storage_properties FROM test.tables WHERE location IS NOT NULL": [
+                make_row(("s3://test_location/test1/table1", ""), select_cols),
+                make_row(("gcs://test_location2/test2/table2", ""), select_cols),
+                make_row(("abfss://cont1@storagetest1.dfs.core.windows.net/test2/table3", ""), select_cols),
+            ],
+        }
+    )
+    location_crawler = ExternalLocations(ws, sbe, "test")
+    ws.external_locations.list.return_value = [ExternalLocationInfo(name="loc1", url="s3://test_location/test11")]
+    location_crawler.save_as_terraform_definitions_on_workspace("~/.ucx")
+    (path, content), _ = ws.workspace.upload.call_args
+    assert "~/.ucx/external_locations.tf" == path
+    assert (
+        'resource "databricks_external_location" "test_location_test1" { \n'
+        '    name = "test_location_test1"\n'
+        '    url  = "s3://test_location/test1"\n'
+        "    credential_name = databricks_storage_credential.<storage_credential_reference>.id\n"
+        "}\n"
+        'resource "databricks_external_location" "test_location2_test2" { \n'
+        '    name = "test_location2_test2"\n'
+        '    url  = "gcs://test_location2/test2"\n'
+        "    credential_name = databricks_storage_credential.<storage_credential_reference>.id\n"
+        "}\n"
+        'resource "databricks_external_location" "storagetest1_test2" { \n'
+        '    name = "storagetest1_test2"\n'
+        '    url  = "abfss://cont1@storagetest1.dfs.core.windows.net/test2"\n'
```

Comment:
Bug: container name is not part of automated name inference. This breaks when many containers are in the same storage account